backend/src/process.py

```diff
@@ -71,7 +72,10 @@ async def __process(self, node: UsableData) -> Any:
         logger.debug(f"Running node {node_id}")
         # Return cached output value from an already-run node if that cached output exists
         if self.output_cache.get(node_id, None) is not None:
-            finish_data = await self.check()
+            finish_data = {
+                "finished": [key for key in self.output_cache.keys()],
+                "broadcastData": self.broadcast_cache,
```

Comment:
You send over **all** data every time something changes?

That's not good. The small image previews were designed to be around 1MB. This is fine for a single image, but not if you have potentially dozens of them.

Reply:
That's why I said it's concerning in my initial comment